src/go/k8s/controllers/redpanda/console_controller.go

```diff
@@ -0,0 +1,232 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package redpanda
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	redpandav1alpha1 "github.com/redpanda-data/redpanda/src/go/k8s/apis/redpanda/v1alpha1"
+	adminutils "github.com/redpanda-data/redpanda/src/go/k8s/pkg/admin"
+	consolepkg "github.com/redpanda-data/redpanda/src/go/k8s/pkg/console"
+	"github.com/redpanda-data/redpanda/src/go/k8s/pkg/resources"
+)
+
+// ConsoleReconciler reconciles a Console object
+type ConsoleReconciler struct {
+	client.Client
+	Scheme                *runtime.Scheme
+	Log                   logr.Logger
+	AdminAPIClientFactory adminutils.AdminAPIClientFactory
+	clusterDomain         string
+	Store                 *consolepkg.Store
+	EventRecorder         record.EventRecorder
+}
+
+const (
+	// Warning event if referenced Cluster not found
+	ClusterNotFoundEvent = "ClusterNotFound"
+
+	// Warning event if subdomain is not found in Cluster ExternalListener
+	NoSubdomainEvent = "NoSubdomain"
+)
+
+//+kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=apps,resources=configmaps,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="",resources=events,verbs=get;list;watch;create;update;patch
+
+//+kubebuilder:rbac:groups=redpanda.vectorized.io,resources=consoles,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=redpanda.vectorized.io,resources=consoles/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=redpanda.vectorized.io,resources=consoles/finalizers,verbs=update
+
+func (r *ConsoleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("redpandaconsole", req.NamespacedName)
+
+	log.Info(fmt.Sprintf("Starting reconcile loop for %v", req.NamespacedName))
+	defer log.Info(fmt.Sprintf("Finished reconcile loop for %v", req.NamespacedName))
+
+	console := &redpandav1alpha1.Console{}
+	if err := r.Get(ctx, req.NamespacedName, console); err != nil {
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+	// Checks if Console is valid to be created in specified namespace
+	// TODO: This should be in validating webhook once finalized if syncing Secret must be done
```

Comment:
I thought of adding this flag to unblock the PR because as @nicolaferraro said it might not be ideal to allow installing Console in different namespace from Redpanda Cluster. But we haven't discussed it yet and @RafalKorepta wants to have this flexibility.

Reply:
This needs more discussion if we need this flag. But setup the webhook as requested :+1: [REF](https://github.com/redpanda-data/redpanda/pull/5758/commits/8b1e0700d86b1614d783215a2803458f3cb1e850) Marking as resolved